### PR TITLE
GREEDY field templates for the graphite parser

### PR DESCRIPTION
**Pull request by chrusty**

### CODE REVIEW

File: plugins/parsers/graphite/parser_test.go [hunk 1]
```diff
@@ -94,6 +94,13 @@ func TestTemplateApply(t *testing.T) {
 			measurement: "cpu.load",
 			tags:        map[string]string{"zone": "us-west"},
 		},
+		{
+			test:        "conjoined fields",
```
Comment (sparrc) [inline plugins/parsers/graphite/parser_test.go]: here you should test that the correct fields get applied.

ie, this should test that measurement=="cpu.util" and field=="idle.percent" (feel free to make a separate test function for testing field templates)

**Comment by sparrc** [comment]
@chrusty can this also support specifying multiple fields?, ie, `host.measurement.measurement.field.field`

**Comment by chrusty** [comment]
@sparrc OK, I've added a test for the field-name. Regarding the multiple-fields question, no at the moment it doesn't, but I could add it.

I found this example in another issue (https://github.com/influxdata/telegraf/issues/39):
```
cpu.usage,host=server01,region=us-west:idle=10.0,user=50.0,system=40.0|g
=> statsd_cpu_usage_gauge,host='server01',region='us-west' idle=10,user=50,system=50```

I'm skiing this week, but i'll see if i get a bad-weather day to take a look.

**Comment by sparrc** [comment]
@chrusty any updates?

**Comment by chrusty** [comment]
@sparrc So the initial part of the PR seems to work fine, but I'm struggling with the "multiple fields" thing... is there an accepted way of sending multi-value metrics with StatsD, or would people just send several metrics instead?

**Comment by sparrc** [comment]
@chrusty that's not what I had in mind exactly, I was thinking a template like this:

```
host.measurement.measurement.field.field
```

would translate:

```
localhost.cpu.usage.idle.percent 99
=> cpu_usage,host=localhost idle_percent=99
```

File: plugins/parsers/graphite/parser_test.go [hunk 2]
```diff
@@ -574,15 +594,48 @@ func TestApplyTemplateField(t *testing.T) {
...
+func TestApplyTemplateMultipleFieldsTogether(t *testing.T) {
 	p, err := NewGraphiteParser("_",
-		[]string{"current.* measurement.field.field"}, nil)
+		[]string{"current.* measurement.measurement.field.field"}, nil)
 	assert.NoError(t, err)
 
-	_, _, _, err = p.ApplyTemplate("current.users.logged_in")
-	if err == nil {
-		t.Errorf("Parser.ApplyTemplate unexpected result. got %s, exp %s", err,
-			"'field' can only be used once in each template: current.users.logged_in")
```
Comment (chrusty) [inline plugins/parsers/graphite/parser_test.go]: I had to remove this constraint to allow the new feature.

**Comment by chrusty** [comment]
@sparrc I think this does what you want. Sorry, I'd assumed that the feature you'd asked for was much more complex!

**Pull request closed without merging**